appendix-basic65.tex

```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
 \end{tabular}
 \end{center}
 
+A variable whose name is a single letter followed by the type symbol (or no
+symbol for real number variables) is a {\em fast variable}. BASIC 65 stores
+the variable in a way that makes it faster to access or update the value than
+variables with longer names. It otherwise behaves like any other variable.
+
 \subsection{BASIC 65 Arrays}
 \index{BASIC 65 Arrays}
 
-Each array consumes the number of elements multiplied by the item size,
-plus the size of the header (6 + 2 * dimensions) in memory.
-For example the array
-\begin{tcolorbox}[colback=black,coltext=white]
-\verbatimfont{\codefont}
-\begin{verbatim}
-100 DIM X(8,2,3)
-\end{verbatim}
-\end{tcolorbox}
-has 3 dimensions and 108 (9 x 3 x 4) items. You might be asking: Why is it 9 x 3 x 4,
-when the program uses 8, 2, and 3? This is because array indexes start at 0, not 1.
-
-The size for real items is 5, so the data of that array above would occupy 540 (5 x 108) bytes.
-The header size is 12 bytes (6 + 2 * 3), so the total length in memory is 552 bytes (540 + 12).
-
-\begin{comment}
-Arrays are stored in bank 1 starting at address \$2000 and expand upwards.
-They share the available memory at \$2000 .. \$F6FF with the string area,
-which starts in bank 1 at address \$F6FF, and expands \textit{downwards}.
-\end{comment}
-Each of the above scalar variable types can be used as an array, by declaring
-them with a {\bf DIM} statement. The arrays are initialised to zero for all
-elements on declaration. If an undeclared array element is used,
-an automatic implicit declaration is performed, which sets the upper boundary
-for each dimension to 10. For example, the usage of an undeclared element
-\screentext{AB(3,5)} would automatically perform a \screentext{DIM AB(10,10)}.
-As noted previously, the lower boundary for each dimension is always 0 (zero),
-so an array initialised with \screentext{DIM AB(10)} consists of 11 elements and accepts indexes from
-0 to 10.
-
-String arrays are more precisely expressed as \textit{arrays of string
```

Comment:
While I know sharing internal details of strings is probably overkill for new users, I have to admit, this tidbit did help me in debugging some of my own issues with strings in the past.

It's one of those tidbits that you don't know you need it until you need it. Just having the words there, letting the new user just skim over them without any great insight, but then at some later time when they're delving into string-matters more deeply, they might have a faint recollection of such a detail getting mentioned in the manual.

Reply:
This is described in POINTER, which is the right place for this for now. (Also in FREAD and FWRITE, which use POINTER in an example of reading and writing variables in a file.)